cpp/src/arrow/array.h

```diff
@@ -29,6 +29,7 @@
 #include "arrow/type_fwd.h"
 #include "arrow/type_traits.h"
 #include "arrow/util/bit-util.h"
+#include "arrow/util/decimal.h"
```

Comment:
You're leaking boost/multiprecision headers in the public API here, can this be removed?